lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -661,10 +670,9 @@ def _purge_users(self):
 
     def _put_admin_user(self, pwd: Optional[str] = None):
         """Change password of Admin user."""
-        is_update = pwd is not None
-        hashed_pwd, pwd = generate_hashed_password(pwd)
-
-        if is_update:
+        # update
+        if pwd is not None:
+            hashed_pwd, pwd = generate_hashed_password(pwd)
```

Comment:
If we don't store the password and hash here, then the`_on_set_password_action` function should store both of them.

Reply:
We store them on success [here](https://github.com/canonical/opensearch-operator/pull/95/files/8643c1c13d920b3b61c10c9c4ae620cd16adb5ac#diff-5c0d89f838ffd5a9ecf2288b3698320fda2d2b2b737361b6e10adb55d1929f43R707-R708)